File: Service/Implement/FactoryService.cs
namespace Service.Implement
{
    public class FactoryService : BaseService<Factory, IFactoryRepository>
    , IFactoryService
    {
        private readonly IFactoryRepository _FactoryRepository;
        public FactoryService(IFactoryRepository FactoryRepository) : base(FactoryRepository)
        {
            _FactoryRepository = FactoryRepository;
        }
    }
}

File: ERP_Angular/src/app/shared/ERP/Factory.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { MatTableDataSource } from '@angular/material/table';
import { environment } from 'src/environments/environment';
import { Factory } from './Factory.model';
import { BaseService } from './Base.service';
import { BaseResult } from 'src/app/shared/ERP/BaseResult.model';

@Injectable({
    providedIn: 'root'
})
export class FactoryService extends BaseService {
    DisplayColumns001: string[] = ['No', 'ID', 'Code', 'Name', 'Display', 'Note', 'SortOrder', 'Active', 'Save'];

    List: Factory[] | undefined;
    ListFilter: Factory[] | undefined;
    FormData!: Factory;
    constructor(public httpClient: HttpClient) {
        super(httpClient);
        this.Controller = "Factory";
    }
}

